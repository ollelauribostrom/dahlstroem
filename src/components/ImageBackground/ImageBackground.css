.image-background {
  position: absolute !important;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  display: flex;
  align-items: center;
  justify-content: center;
}

.image-background__img {
  height: 80vh;
}

@media only screen and (max-device-width: 1024px) {
  .image-background__img {
    height: 60vh;
  }
}
