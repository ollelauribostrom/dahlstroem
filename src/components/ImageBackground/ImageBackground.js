import Img from 'gatsby-image';
import React, {useState} from 'react';
import {StaticQuery, graphql} from 'gatsby';

const ImageBackground = () => {
  const [isLoaded, setIsLoaded] = useState (false);
  return (
    <StaticQuery
      query={graphql`
        query GifQuery {
          datoCmsSetting {
            gif {
              url
            }
            gifThumbnail {
              fluid(
                maxWidth: 1280
                imgixParams: { fm: "jpg", auto: "compress" }
              ) {
                ...GatsbyDatoCmsSizes
              }
            }
          }
        }
      `}
      render={data => {
        const {gif, gifThumbnail} = data.datoCmsSetting;

        return (
          <div className="image-background">
            <Img
              fluid={gifThumbnail.fluid}
              className="image-background__img"
              style={{display: isLoaded ? 'hidden' : 'block'}}
            />
            <img
              decoding="async"
              src={gif.url}
              className="image-background__img"
              style={{display: isLoaded ? 'block' : 'hidden'}}
              onLoad={() => setIsLoaded (true)}
            />
          </div>
        );
      }}
    />
  );
};

export default ImageBackground;
