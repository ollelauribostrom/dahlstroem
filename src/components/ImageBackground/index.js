import ImageBackground from './ImageBackground';
import './ImageBackground.css';

export default ImageBackground;
