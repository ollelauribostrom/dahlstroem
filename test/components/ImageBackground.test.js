import React from 'react';
import { shallow } from 'enzyme';

import ImageBackground from '../../src/components/ImageBackground';

describe('Tests for ImageBackground', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ImageBackground
        src={{
          fluid: {
            aspectRatio: 1.4678899082568808,
            sizes: '(max-width: 1800px) 100vw, 1800px',
            src: 'someurl',
            srcSet: 'someurl',
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
