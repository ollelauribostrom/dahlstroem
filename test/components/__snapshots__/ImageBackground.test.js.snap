// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ImageBackground renders correctly 1`] = `
<Image
  Tag="div"
  alt=""
  className="imagebackground"
  critical={false}
  fadeIn={true}
  fluid={
    Object {
      "aspectRatio": 1.4678899082568808,
      "sizes": "(max-width: 1800px) 100vw, 1800px",
      "src": "someurl",
      "srcSet": "someurl",
    }
  }
/>
`;
